### PR TITLE
Exposed sd folder as a volume, updated dependencies to be fully compliant with CUDA 12

### PR DESCRIPTION
**Pull request by andy-kru**
Hey @rgryta, thank you for your project! I was using it for some time to play with AUTOMATIC1111 and made some changes for myself that can be useful:
1. Exposed stable diffusion folder as a volume in WSL to simplify interaction with AUTOMATIC1111. Models/LoRAs/styles/etc. can be copied on the fly to `/home/sd/stable-diffusion-webui` WSL folder without restarting the docker container or rebuilding the image. I changed a bit the provisioning part to separate sample models from the docker image (the image will contain only fresh AUTOMATIC1111 without any models). I also reuse a single container by default to save state/user settings/extensions.
2. Dependencies update. Recently, PyTorch and xFormers libraries were released with the support of CUDA 12.1 (previously, they were compiled for 11.8 only), so I have updated scripts to use specific 12.1 versions of these libraries and also updated python to 3.11.

### CODE REVIEW

**Comment by rgryta** [comment]
Hey, thanks for your contribution.
I'll test and review it in the coming days - it's quite a busy week so I may have to push it until the next one though.

File: install/install.ps1 [hunk 1]
```diff
@@ -76,17 +76,21 @@ $ignr = wsl -d $distro -u root -e sh -c 'systemctl restart docker'
...
 Write-Host 'Now local container will be built. This can take about an hour depending on your CPU and internet speed.'
 wsl -d $distro -e sh -c "cd ``wslpath -a '$scriptPath'``/../docker && ./build.sh"
 
-Clear-Host
-Write-Host 'You can now use `run.bat` to launch Stable Diffusion. Closing this window in 5 seconds...'
+wsl -d $distro -e sh -c "mkdir -p /home/sd/stable-diffusion-webui"
+wsl -d $distro -e sh -c "docker volume create --driver local --opt type=none --opt device=/home/sd/stable-diffusion-webui --opt o=bind sd_vol"
+wsl -d $distro -e sh -c "docker create -v sd_vol:/home/sd/stable-diffusion-webui -p 127.0.0.1:7860:7860 --name sd --gpus all sd"
+
+wsl -d $distro -e sh -c "./provision.sh"
```
Comment (rgryta) [inline install/install.ps1:93]: During installation PowerShell is ran as admin. As such you have to provide full path to the script if you'd like to run it, similar to line #87:
```
wsl -d $distro -e sh -c "cd ``wslpath -a '$scriptPath'``/../docker && ./build.sh"
```
At the moment installation script fails on this step.

File: install/webui-user.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
```
Comment (rgryta) [inline install/webui-user.sh:1]: Is this file used anywhere? It's probably meant to be an override to the default webui-user.sh contained in AUTOMATIC1111 project - if so, please move it under docker/prep and ensure it's overwriting it in Dockerfile.

Correct me though if my understanding there is wrong.

**Review by rgryta — CHANGES_REQUESTED** [review]
Thanks again for the PR.

Could you maybe provide a list of differences between default UI settings and those provided by the attached config.json - I'd also appreciate appending them to README as a separate section.

**Comment by andy-kru** [comment]
> Thanks again for the PR.
> 
> Could you maybe provide a list of differences between default UI settings and those provided by the attached config.json - I'd also appreciate appending them to README as a separate section.

Hey, I have cleaned up `config.json` to include only delta changes:
- `sd_vae`: to preselect default VAE file. I saw some issues with `--vae-path` command line argument that you used, as it's not working properly when you want to disable VAE.
- `quicksettings_list`: to add SD VAE selector, so user can manage VAE configuration

image

File: install/install.ps1 (hunk 1 above)
Comment (andy-kru) [reply install/install.ps1]: Thanks, fixed. I used non-standart command before to run `install.ps1`, now it should work.

File: install/webui-user.sh (hunk 2 above)
Comment (andy-kru) [reply install/webui-user.sh:1]: Yeah, this file is used on provision step (install.ps1:93) to add custom command line arguments and . I left it in `install` folder as it's not included in Docker image and copied directly to the container (so user can change command line parameters without rebuilding Docker image). If you want, I can move it to `docker/prep`

**Review by rgryta — APPROVED** [review]

**Comment by rgryta** [comment]
Hi @andy-kru, took a while, but I've merged your suggestions.
Functionality is preserved as it was, which is great, I'd appreciate it if you could write a README section for a random user with simple instructions for how to interact with the directory that stores the models (it's obviously not critical, since everything works just fine out of the box with the base xl model).

**Pull request merged**